### PR TITLE
close will use reason file for SIGUSR2 kill

### PR DESCRIPTION
**Pull request by bill1600**
When killed with SIGUSR2, parodus will look for a reason file specified by cmd line parameter, and use the text in that file for the close reason.

### CODE REVIEW

File: src/connection.c [hunk 1]
```diff
@@ -808,18 +808,35 @@ static noPollConnOpts * createConnOpts (char * extra_headers, bool secure)
...
+		char *sigusr2_reason;
+        if (readFromFile (get_parodus_cfg()->close_reason_file, &sigusr2_reason)) {
+          nopoll_conn_close_ext(conn, CloseNormalClosure, sigusr2_reason, 
+			strlen (sigusr2_reason));
+		  ParodusInfo ("Closed by SIGUSR2, reason: %s\n", sigusr2_reason);
+		  free (sigusr2_reason);
+		  return;
+		}
+		/* file could not be read. use canned message. */
+        effective_reason = "restarted with SIGUSR2";
```
Comment (shilpa24balaji) [inline src/connection.c]: @schmidtw what hardcoded default reason would you prefer for the SIGUSR2 restart case?
These are the current values:
parodus_stopping - default, system_restarting - SIGUSR1

File: src/ParodusInternal.c [hunk 2]
```diff
@@ -131,6 +131,27 @@ char* getWebpaConveyHeader()
...
+	if (fp == NULL)
+	{
+		ParodusError("Failed to open file %s (errno %d)\n", file_name, errno);
+		return 0;
+	}
+	fseek(fp, 0, SEEK_END);
+	ch_count = ftell(fp);
+	fseek(fp, 0, SEEK_SET);
+	*data = (char *) malloc(sizeof(char) * (ch_count + 1));
+	fread(*data, 1, ch_count,fp);
```
Comment (shilpa24balaji) [inline src/ParodusInternal.c]: @schmidtw We are assuming the close-reason-file will just have the reason in plain text without any format.
Is this fine or should we use json file with key "reason"?

Comment (schmidtw) [reply src/ParodusInternal.c]: Just plain text makes sense.

**Comment by shilpa24balaji** [comment]
Not required

**Pull request closed without merging**